### ISSUE TITLE
`torch.distributions.Categorical(logits=...).sample()` returns -9223372036854775808 on `MPS`. Works correctly on `CPU` backend.

### ISSUE THREAD

**Issue by nipunbatra** [pytorch/pytorch#148949]
### 🐛 Describe the bug

```python
import torch
device = 'cpu'
t = torch.tensor([-0.6194,  0.2150,  0.0741, -0.5155, -0.3574,  0.1880,  0.3493,  0.2933,
          0.3222,  0.1351, -0.1676,  0.2195, -0.2661, -0.1681,  0.0102, -0.2942,
          0.1377, -0.3102,  0.0231, -0.3813, -0.8353, -0.0413, -0.2836, -0.0108,
         -0.6760, -0.0350, -0.6092], device=device)
print(torch.distributions.Categorical(logits=t).sample())

device = 'mps'
t = torch.tensor([-0.6194,  0.2150,  0.0741, -0.5155, -0.3574,  0.1880,  0.3493,  0.2933,
          0.3222,  0.1351, -0.1676,  0.2195, -0.2661, -0.1681,  0.0102, -0.2942,
          0.1377, -0.3102,  0.0231, -0.3813, -0.8353, -0.0413, -0.2836, -0.0108,
         -0.6760, -0.0350, -0.6092], device=device)
print(torch.distributions.Categorical(logits=t).sample())
```

Output

```python
tensor(18)
tensor(-9223372036854775808, device='mps:0')
```

The code clearly works correctly on CPU backend but doesn't work correctly on MPS backend.

This value `-9223372036854775808` is apparently the minimum value of `int64` data type.

### Versions

```bash
Collecting environment information...
PyTorch version: 2.0.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.3.1 (x86_64)
GCC version: Could not collect
Clang version: 13.0.1
CMake version: version 3.23.2
Libc version: N/A

Python version: 3.9.20 | packaged by conda-forge | (main, Sep 30 2024, 17:51:21)  [Clang 17.0.6 ] (64-bit runtime)
Python platform: macOS-15.3.1-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] gpytorch==1.9.1
[pip3] hamiltorch==0.4.1
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.26.4
[pip3] pytorch-lightning==1.9.3
[pip3] pytorch-metric-learning==1.7.3
[pip3] torch==2.0.0
[pip3] torchaudio==0.12.1
[pip3] torchmetrics==0.8.2
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.15.1
[conda] autopytorch               0.2.1                    pypi_0    pypi
[conda] gpytorch                  1.9.1                    pypi_0    pypi
[conda] hamiltorch                0.4.1                    pypi_0    pypi
[conda] mkl                       2022.2.1         h44ed08c_16952    conda-forge
[conda] mkl-service               2.4.0            py39h9032bd8_0    conda-forge
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.26.4           py39h28c39a1_0    conda-forge
[conda] pytorch-lightning         1.9.3                    pypi_0    pypi
[conda] pytorch-metric-learning   1.7.3                    pypi_0    pypi
[conda] torch                     2.0.0                    pypi_0    pypi
[conda] torchaudio                0.12.1                 py39_cpu    pytorch
[conda] torchmetrics              0.8.2                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.15.1                   pypi_0    pypi

```

cc @kulinseth @albanD @malfet @DenisVieriu97 @jhavukainen

**Comment by Isalia20** [comment]
Can you try with a more recent torch version? Can't reproduce it with the latest nightly build:
```
tensor(1)
tensor(18, device='mps:0')
```
